usage instruction.md

```diff
@@ -1,13 +1,47 @@
-本仓库用于leetcode刷题记录，包括题目、解答和注释等内容。
+This repository serves as a record of my LeetCode coding exercises, including problem descriptions, solutions, and annotations.
 
-题目源于https://leetcode.cn/problemset/all/ ，主要采用C/C++语言编程，在leetcode自带编辑器上运行。
+The problems are sourced from [leetcode](https://leetcode.com/problemset/all/) and are primarily solved using the C/C++ programming language within the LeetCode editor.
 
-题目分为简单、中等和困难三类文件夹，文件夹中每一个文件均以题号_题目名称命名。
+The problems are categorized into three folders: Easy, Medium, and Hard. Each folder contains individual files named with the problem number and name format: "ProblemNumber_ProblemName".
 
-文件内容以#开头，是题目的具体描述及示例，接下来是已通过的代码、文字注释和思路记录。
+The file content begins with a "#" followed by the problem's description and examples. It is then followed by the code, comments, and the thought process for the solutions.
 
-每周创建新分支以weekN_20xx/xx命名，分别记录练习的第几周以及具体时间，每周日合并上传到main分支，并同步到本地仓库。
+A new branch is created each week named "weekN_20xx/xx", indicating the week number and specific dates of the exercise.The changes are merged into the main branch and synchronized with the local repository on Sundays.
 
-您可以通过右上角搜索框搜索题号或题目具体查询解答过程，若发现某题解答思路有误，可联系邮箱qianyi0830@163.com，感谢。
+## For Windows Users
+1. Make sure you have an appropriate compiler installed on the Windows system. You can choose to use Visual Studio Build Tools or CodeBlocks (which comes with GCC).
+2. If you choose to use Visual Studio Build Tools, refer to the official documentation for downloading and installation instructions.
+   If you choose to use CodeBlocks, you can download the latest version from the official website and follow the installation wizard.
+3. Open your code file in CodeBlocks and click the compile and run.
+   
+## For Mac Users
+1. Make sure you have Xcode Command Line Tools installed on your Mac. If not, you can install them by running the following command in the terminal:
+   ```
+   xcode-select --install
+   ```
+2. In the terminal, navigate to your local repository folder.
+3. Use the following commands to compile and run the code:
+   ```
+   gcc file_name.c -o executable_name
+   ./executable_name
+   ```
+   
+## For Linux Users
+1. In the terminal, install an appropriate compiler using the package manager. Common choices include GCC, Clang, or LLVM.
+    You can install GCC using the following commands:
```

Comment:
Please be noted that `apt` is usually available on the Debian family (mostly famous for Ubuntu), not the others (like RPM-based Red Hat, CentOS, or Fedora, or those very tiny ones like BusyBox).